### PR TITLE
Refresh and rework requirements

### PR DESCRIPTION
**Pull request by alimanfoo**
This PR reworks the requirements files for easier maintenance:

* All indirect dependencies have been removed from requirements files.
* File requirements_dev.txt contains only direct requirements needed to run numcodecs, with versions pinned. This is something we would track with a bot to automatically upgrade.
* New file requirements_test.txt contains packages needed to run tests, not pinned. I have not pinned these because otherwise I fear the churn from a bot would get too much. 

TODO:
* [ ] ``tox -e py37`` passes locally
* [ ] ``tox -e py27`` passes locally
* [ ] Changes documented in docs/release.rst
* [ ] ``tox -e docs`` passes locally
* [ ] AppVeyor and Travis CI passes
* [ ] Test coverage to 100% (Coveralls passes)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,7 +22,6 @@ matrix:
 
 install:
   - pip install -U tox-travis coveralls pip setuptools wheel
-  - pip install -r requirements_dev.txt
```
Comment (alimanfoo) [inline .travis.yml]: This is not actually needed as installation is handled within tox.

File: numcodecs/tests/common.py [hunk 2]
```diff
@@ -12,7 +12,7 @@
 
 
 from numcodecs.compat import buffer_tobytes, ndarray_from_buffer
-from numcodecs import *  # flake8: noqa
+from numcodecs.registry import get_codec
```
Comment (jakirkham) [inline numcodecs/tests/common.py]: Think this might be related to the CI failure. Not sure though.

**Comment by jakirkham** [comment]
Generally seems reasonable to me.

File: numcodecs/tests/common.py (hunk 2 above)
Comment (alimanfoo) [reply numcodecs/tests/common.py]: Yep, my bad. Should be fixed now.

**Comment by alimanfoo** [comment]
Thanks @jakirkham. I'll wait for appveyor to pass then merge and setup PyUP if no objections.

**Comment by jakirkham** [comment]
As long as it's green, I'm happy. 😄

**Pull request merged**